RNWCPP/ReactUWP/Views/ViewViewManager.cpp

```diff
@@ -123,26 +131,41 @@ class ViewShadowNode : public ShadowNodeBase
     TabIndex(TabIndex());
   }
 
-  ViewPanel* GetViewPanel()
+  winrt::com_ptr<ViewPanel> GetViewPanel()
   {
+    XamlView current = m_view;
+
     if (IsControl())
     {
-      auto contentControl = m_view.as<winrt::ContentControl>();
-      return contentControl.Content().as<ViewPanel>().get();
+      auto control = m_view.as<winrt::ContentControl>();
+      current = control.Content().as<XamlView>();
     }
-    else
+
+    if (HasOuterBorder())
     {
-      return m_view.as<ViewPanel>().get();
+      auto border = current.try_as<winrt::Border>();
+      current = border.Child().try_as<XamlView>();
     }
+
+    auto panel = current.try_as<ViewPanel>();
+    assert(panel != nullptr);
+
+    return std::move(panel);
   }
 
   winrt::impl::com_ref<react::uwp::ViewControl> GetControl()
   {
-    return IsControl() ? m_view.as<react::uwp::ViewControl>() : nullptr;
+    auto control = m_view.try_as<ViewControl>();
+    auto panel = m_view.try_as<ViewPanel>();
+    auto border = m_view.try_as<winrt::Border>();
+
```

Comment:
these seem to be unreferenced locals?

Reply:
Yeah, I'd already removed that in my branch (debug code).